### ISSUE TITLE
Report useless ternary operator

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1706]
# Rule request

## Thesis

Detect if both possible results of a ternary operator return the same value.

```python
# bad
# (all results below can be replaced by just `a`)
a if ... else a
a if a is not None else None
a if a != b else b
b if a == b else a

# ok
a if ... else c
a.split() if a is not None else None
a if a != b else c
```

## Reasoning

It is either a bug or unnecessary complexity.

**Comment by locko62** [comment]
Hello word

**Comment by sobolevn** [comment]
@locko62 does it mean that you want to wrok on this issue? 🙂

**Comment by Praveenk8051** [comment]
Hello, I would like to work on this? This is my first-time!

**Comment by sobolevn** [comment]
@Praveenk8051 you are welcome!

**Comment by Praveenk8051** [comment]
Thank you!

**Comment by Praveenk8051** [comment]
Hi, I need some help. 
I went through documentation to know how to contribute and [API References](https://wemake-python-stylegui.de/en/latest/pages/api/index.html#api-reference)

I'm facing below error when I'm executing `make test`. Can you please help me with it ? @sobolevn 
![wemake](https://user-images.githubusercontent.com/20404922/98668565-58da7600-2350-11eb-9405-823159fe7c11.png)


Thanks!

**Comment by sobolevn** [comment]
I can recommend to remove your `venv`, make sure that your base `python` does not have any installed packages that might be causing this problem (check with `pip freeze`, it should be empty), and call `poetry install` once again after cleaning everything.

**Comment by Praveenk8051** [comment]
Worked. Thank you!

**Comment by Praveenk8051** [comment]
Hello, I'm trying to debug and apply changes so that `linter` is applicable properly for ternary properties. But I'm failing to understand how to proceed. Can you please help me with it ? @sobolevn 

This happens to be present in `compares.py` where `NestedTernaryViolation` is checked. Am I on right path ? Should i check and tweak in `compares.py`

**Comment by sobolevn** [comment]
Yes, please fix all the violations in our own source code found by a new rule!

**Comment by amansr02** [comment]
Hello @sobolevn. I would also like to work on this issue apart from #1974 with @bhargavi247. We are students who are working on an assignment to fix Issues for open source projects. Do let us know if this issue is still open! Thank you!

**Comment by sobolevn** [comment]
Yes, it is! Thanks!

**Comment by amansr02** [comment]
Awesome! Let us get started on this issue as well. If you could assign it to @bhargavi247 that would be great.

**Comment by bhargavi247** [comment]
Thanks @sobolevn ! Would love to get assigned to this task :)

**Comment by harshakhmk** [comment]
I would like to work on this issue, I am a first time contributor

**Comment by sobolevn** [comment]
Hi @harshakhmk!
Please, take a look at https://github.com/wemake-services/wemake-python-styleguide/blob/master/CONTRIBUTING.md

**Comment by mketer1** [comment]
Hi! I'm fairly new to open source contribution but would like to give this issue a try. If this issue is unclaimed, could I possibly be assigned. Also, if I do need assistance, would I be able to receive guidance from other contributors in this repository?

**Comment by adlizone** [comment]
This is hot....does anyone have the idea how to write a visitor for this

**Comment by adlizone** [comment]
#2579

**Comment by FieldMarshallObvious** [comment]
Is this issue still open? If so I would be interested in being assigned to this issue. Like many others here also new to open source and would appreciate the opportunity to learn.

**Comment by sobolevn** [comment]
@FieldMarshallObvious please, go ahead :)

**Comment by sobolevn** [comment]
you can take a look at https://github.com/wemake-services/wemake-python-styleguide/pull/2579 
it was a promising start, but never fixed tests :(

**Comment by FieldMarshallObvious** [comment]
Thanks! Will do 👍

**Comment by FieldMarshallObvious** [comment]
Attempt to fix this on PR #2902

**Comment by CodeCraftsman481** [comment]
Hi, I am new to contributing, Can I take this task up?

**Comment by sobolevn** [comment]
@FieldMarshallObvious is working on it right now, please take some other issue :)
There are multiple ones that are free!

**Comment by CodeCraftsman481** [comment]
@sobolevn Can you please suggest other issues for beginners?

**Comment by Aanchallllll** [comment]
Is this issue still relevant? if so, I would like to work on it
